### PR TITLE
jsdoc output

### PR DESCRIPTION
**Pull request by gkatsev**
Uses the TUI JSDoc template. All configuring of jsdoc was moved to the jsdoc.json file.
There's currently a bug with TUI regarding the tutorials (https://github.com/nhnent/tui.jsdoc-template/issues/12) but it should via gh-pages currently because it'll now autorender markdown into html.

Left to do:
* [x] run jsdoc on publish
* [x] tweak configuration of jsdoc
* [x] make sure that tutorials are rendering correctly
* [x] figure out what to do about TOCs in jsdoc
* [ ] fix jsdoc's README

### CODE REVIEW

**Comment by BrandonOCasey** [comment]
Do we want to do custom styling for video.js in another PR?

**Comment by BrandonOCasey** [comment]
Can we change the logo in the top left? (from toastui to videojs)

**Comment by gkatsev** [comment]
Yes, I believe it's all customizable in some way or another. I'll do it in this PR if it isn't too complicated.

**Comment by BrandonOCasey** [comment]
I also noticed that some of the links are broken, I think it would be good to have some sort of automated tests to check for broken links (maybe in another PR as well)

**Comment by BrandonOCasey** [comment]
I also noticed that examples seem to be cut off at the bottom? (might be a bug in tui)
http://gkatsev.github.io/video.js/docs/api/tutorial-components.html

**Comment by gkatsev** [comment]
the tutorials are loaded in in-an iframe, so, you should be able to just scroll :), but yeah, the iframe is potentially not sized correctly.

Also, which links do you see broken?

**Comment by BrandonOCasey** [comment]
the Guides link at http://gkatsev.github.io/video.js/docs/api/index.html leading to http://gkatsev.github.io/video.js/docs/api/guides. Should probably be `/guides`? I guess its not up yet though.

**Comment by BrandonOCasey** [comment]
I also noticed that the API link at the top doesn't just link to the current page.

**Comment by gkatsev** [comment]
Ah, I think that's because it's just inlining the `docs/api.md` file as a "main" and the links are still relative to the markdown structure. That's something that we'd want to fix. We potentially could just write a different page to use as the front of the docs site.

**Comment by gkatsev** [comment]
The latest updates depend on these changes: https://github.com/nhnent/tui.jsdoc-template/pulls/gkatsev

**Comment by gkatsev** [comment]
Looks like to get tocs to work, we'd need to update the template to add [a custom renderer to marked to add embedded anchor tags for headings](https://github.com/chjj/marked#overriding-renderer-methods), I'm going to call that not a blocker.

The only real update that's still necessary and I think we should do it separately from this, probably, is to update the index.md so we have a better output on the main docs page.

File: package.json [hunk 1]
```diff
@@ -25,10 +25,12 @@
     "start": "grunt dev",
     "test": "grunt test",
     "docs": "npm run docs:lint && npm run docs:api",
-    "docs:api": "jsdoc -r src/js -d docs/api -c .jsdoc.json",
+    "jsdoc": "jsdoc",
```
Comment (BrandonOCasey) [inline package.json]: this will only cause "There are no input files to process.", maybe we should skip this script?

File: package.json [hunk 2]
```diff
@@ -25,10 +25,12 @@
...
+    "docs:api": "jsdoc -c .jsdoc.json",
     "docs:lint": "remark -- './**/*.md'",
     "docs:fix": "remark --output -- './**/*.md'",
-    "babel": "babel src/js -d es5"
+    "babel": "babel src/js -d es5",
+    "prepublish": "not-in-install && npm run docs:api || in-install"
```
Comment (BrandonOCasey) [inline package.json]: If I understand this right this script will only build the docs `prepublish` but not when `npm install` is run on the project. Is that right? (if so this is awesome, and I had no idea that we could do this)

File: .jsdoc.json [hunk 3]
```diff
@@ -1,4 +1,38 @@
...
+  "templates": {
+    "default": {
+      "staticFiles": {
+        "include": ["build/docs/"]
+      }
+    },
+    "logo": {
+      "url": "//videojs.com/img/logo.png",
+      "height": "30px",
+      "width": "214px"
```
Comment (BrandonOCasey) [inline .jsdoc.json]: Is there anyway to include videojs.com as a link somewhere on the navigation?

File: .jsdoc.json [hunk 4]
```diff
@@ -1,4 +1,38 @@
...
+    "encoding": "utf8"
+  },
+  "templates": {
+    "default": {
+      "staticFiles": {
+        "include": ["build/docs/"]
+      }
+    },
+    "logo": {
+      "url": "//videojs.com/img/logo.png",
```
Comment (BrandonOCasey) [inline .jsdoc.json]: should this logo url have http:// on the front? doesn't seem to load for me without it.

File: package.json (hunk 2 above)
Comment (gkatsev) [inline package.json]: Yup, that's what this does.

File: .jsdoc.json (hunk 4 above)
Comment (gkatsev) [inline .jsdoc.json]: are you loading it via file:///? That won't work then. But we could potentially just do http://

File: .jsdoc.json (hunk 3 above)
Comment (gkatsev) [inline .jsdoc.json]: Hm... I can definitely add a link on the footer. Making the logo a link would be awesome but might need an update to the template.

File: package.json (hunk 1 above)
Comment (gkatsev) [inline package.json]: This is like the `grunt` task above, it's so that you can more easily run `jsdoc` commands on the cli without installing it globally if `jsdoc -c .jsdoc.json` doesn't suit what you're trying to do.

File: .jsdoc.json (hunk 4 above)
Comment (BrandonOCasey) [inline .jsdoc.json]: without http:// on the front chrome seems to default to file://. Even with that when I open the generated index.html file the logo is not loaded.

File: .jsdoc.json (hunk 3 above)
Comment (BrandonOCasey) [inline .jsdoc.json]: ahh didn't notice the footer link maybe add an issue to add a link in for the logo to the website.

File: package.json (hunk 1 above)
Comment (BrandonOCasey) [inline package.json]: seems outside of the scope of what scripts should do. I think its easy enough to do `./node_modules/.bin/jsdoc`

File: .jsdoc.json (hunk 3 above)
Comment (gkatsev) [inline .jsdoc.json]: I'll add it as a note in 6.0 project.

File: package.json (hunk 1 above)
Comment (gkatsev) [inline package.json]: It's a pretty common pattern nowadays.
